onadata/apps/logger/models/instance.py

```diff
@@ -316,9 +316,22 @@ def update_xform_submission_count_delete(sender, instance, **kwargs):
         safe_delete(f"{DATAVIEW_COUNT}{xform.pk}")
         safe_delete(f"{XFORM_COUNT}{xform.pk}")
 
-        if xform.instances.exclude(geom=None).count() < 1:
-            xform.instances_with_geopoints = False
-            xform.save()
+        # update xform if no instance has geoms
+        if (
+            instance.xform.instances.filter(
+                deleted_at__isnull=True, geom=None
```

Comment:
This query won't do what is expected; It's counts instances that are not deleted with no geoms... Let's use the initial query `xform.instances.exclude(geom=None, deleted_at__isnull=False).count() < 1`

Reply:
`xform.instances.filter(~Q(geom=None), deleted_at__isnull=True).count() < 1`